templates/CRM/Contribute/Form/Contribution/PremiumBlock.tpl

```diff
@@ -332,8 +332,6 @@
           $('#selectProduct').rules('add', 'premiums');
         });
 
-        // need to use jquery validate's ignore option, so that it will not ignore hidden fields
-        CRM.validate.params['ignore'] = '.ignore';
```

Comment:
I can't for the life of me see a field with the ignore class attached. if we want to ignore it I would suggest setting the .crm-no-validate class instead

Reply:
I see what you mean - before this line the `ignore` member is `".select2-offscreen, [readonly], :hidden:not(.crm-select2), .crm-no-validate"`, so it gets overwritten.

The way I read the comment above it it's not so much that there is a field with `.ignore`, it's that at the time it was added there was nothing already in ignore, and so by default it would skip hidden fields if you don't set it to _something_ (https://jqueryvalidation.org/validate/#ignore - `default: ":hidden"`). So what I'm wondering is if you remove it, are they any paths through here where `ignore` might not be set at all, since then `:hidden` fields wouldn't get validated. It might be safer to check if it's undefined first, and if so set it to `.ignore`.

Reply:
@demeritcowboy maybe possibly but also seems to only affect front end contribution page, the add product form and also the backoffice add contribution forms by the looks of https://github.com/civicrm/civicrm-core/search?q=PremiumBlock.tpl&unscoped_q=PremiumBlock.tpl

Reply:
Ok I see - and those other forms don't even have context="makeContribution" so it doesn't even apply there. It's just the contribution form.